pkg/collector/osm_collector_test.go

```diff
@@ -0,0 +1,50 @@
+package collector
+
+import (
+	"os"
+	"testing"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	appsv1 "k8s.io/api/apps/v1"
+)
+
+func TestOsmCollector(t *testing.T) {
+	tests := []struct {
+		name    string
+		want    int
+		wantErr bool
+		deployments []*appsv1.Deployment
```

Comment:
💡 Idea: if you add `collectorName` like in other tests you could test the `GetName()` as well.

so add `collectorName string` here then, below  `value` --> `collectorName "osm"` later under the the test just do this around `line 47`: 

```
        name := c.GetName()
	if name != tt.collectorName {
		t.Errorf("GetName()) = %v, want %v", name, tt.collectorName)
	}
```